tiledb/libtiledb.pyx

```diff
@@ -5044,6 +5043,15 @@ cdef class FileHandle(object):
 
     def __dealloc__(self):
         cdef Ctx ctx = self.vfs.ctx
+        cdef int rc = TILEDB_OK
+        cdef int isclosed = 0
+        check_error(ctx,
+                    tiledb_vfs_fh_is_closed(ctx.ptr, self.ptr, &isclosed))
+        if not isclosed:
+            with nogil:
```

Comment:
@jakebolewski

Reply:
This seems like the behavior we want, need to be careful about resource use here:
https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#controlling-deallocation-and-garbage-collection-in-cpython